rich/_emoji_replace.py

```diff
@@ -1,11 +1,16 @@
-from typing import Match
+from typing import Callable, Match
 
 import re
 
 from ._emoji_codes import EMOJI
 
 
-def _emoji_replace(text: str, _emoji_sub=re.compile(r"(:(\S*?):)").sub) -> str:
+_EmojiSub = Callable[[Callable[[Match[str]], str], str], str]
```

Comment:
This was hard to parse in my head, but I double checked it and it's entirely correct. Bravo.

The thing is that readability has been thrown out for the sake of correctness, which I'm no fan of. Could we break this up a bit to make it more readable? Something along these lines:

```python
_ReStringMatch = Match[str]  # regex match object
_ReSubCallable = Callable[[_ReStringMatch], str] # Callable invoked by re.sub
_EmojiSubMethod = Callable[_ReSubCallable, str]. # Sub method of a compiled re
```